schema.cc

```diff
@@ -1051,6 +1055,8 @@ schema_builder& schema_builder::with_view_info(utils::UUID base_id, sstring base
 }
 
 schema_builder& schema_builder::with_index(const index_metadata& im) {
+    _is_index = true;
```

Comment:
Does this work after a reboot, when the tables are reloaded from disk and not created from scratch? 
I thought you might need to modify `view_info::initialize_base_dependent_fields()` to figure out the right is_index from information in its base after a load. But I have to admit I am not very familiar with this code, or what exactly happens when reading schemas from disk. Please test this (you can add printouts to where you set _is_index = true, and see if it actually gets run during boot).